integration-tests/common/com.liferay.ide.core.base/src/com/liferay/ide/core/base/action/EnvAction.java

```diff
@@ -0,0 +1,259 @@
+			retval = System.getenv("USER");
+		}
+
+		Assert.assertTrue((retval != null) && !retval.equals(""));
+
+		return retval;
+	}
+
+	public File getValidationDir() {
+		IPath path = getBundlesPath().append("validation");
+
+		return path.toFile();
+	}
+
+	public ValidationMsg[] getValidationMsgs(File csv) {
+		Assert.assertTrue(csv.exists());
+
+		String[][] msgs = CSVReader.readCSV(csv);
+
+		ValidationMsg[] validationMsgs = new ValidationMsg[msgs.length];
+
+		for (int i = 0; i < msgs.length; i++) {
+			validationMsgs[i] = new ValidationMsg();
+
+			String[] columns = msgs[i];
+
+			for (int t = 0; t < columns.length; t++) {
+				if (t == 0) {
+					validationMsgs[i].setInput(columns[t]);
+				}
+				else if (t == 1) {
+					validationMsgs[i].setExpect(columns[t]);
+				}
+				else if (t == 2) {
+					validationMsgs[i].setOs(columns[t]);
+				}
+			}
+		}
+
+		return validationMsgs;
+	}
+
+	public boolean internal() {
+		boolean retval = false;
+
+		if ((_internal == null) || _internal.equals("") || _internal.equals("null")) {
+			retval = true;
+		}
+		else {
+			retval = Boolean.parseBoolean(_internal);
+		}
+
+		boolean reachable = false;
+
+		if (retval) {
+			try {
+				InetAddress address = InetAddress.getByAddress(_internalServerIp);
+
+				reachable = address.isReachable(2000);
+			}
+			catch (Exception e) {
+			}
+			finally {
+				Assert.assertTrue("The argument \"internal\" is true but can't reach the internal server", reachable);
+			}
+		}
+
+		return retval;
+	}
+
+	public boolean localConnected() {
+		boolean connected = true;
+
+		try {
+			URL url = new URL("http://127.0.0.1:8080");
+
+			url.openStream();
+		}
+		catch (Exception e) {
+			connected = false;
+		}
+
+		return connected;
+	}
+
+	public boolean notInternal() {
+		return !internal();
+	}
+
+	public final String test_in_the_internal_net =
+		"Only do this test in the internal net, add -Dinternal=\"false\" if you are out of the China office.";
+
+	private EnvAction() {
+		_bundleInfos = _parseBundleInfos();
+	}
+
+	private BundleInfo[] _parseBundleInfos() {
+		IPath bundlesCsvPath = getBundlesPath().append("bundles.csv");
+
+		File bundleCsv = bundlesCsvPath.toFile();
+
+		Assert.assertTrue(bundleCsv.exists());
+
+		String[][] infos = CSVReader.readCSV(bundleCsv);
+
+		BundleInfo[] bundleInfos = new BundleInfo[infos.length];
+
+		for (int i = 0; i < infos.length; i++) {
+			bundleInfos[i] = new BundleInfo();
+
+			String[] columns = infos[i];
+
+			for (int t = 0; t < columns.length; t++) {
+				String value = columns[t];
+
+				if (t == 0) {
+					bundleInfos[i].setBundleZip(value);
+				}
+				else if (t == 1) {
+					bundleInfos[i].setBundleDir(value);
+				}
+				else if (t == 2) {
+					bundleInfos[i].setServerDir(value);
+				}
+				else if (t == 3) {
+					bundleInfos[i].setType(value);
+				}
+				else if (t == 4) {
+					bundleInfos[i].setVersion(value);
+				}
+			}
+		}
+
+		return bundleInfos;
+	}
+
+	private static EnvAction _envAction;
```

Comment:
Why do we need a static instance?  can this not just be made static utilities method?    If there is `state` that needs to be kept, lets make this a junit test rule that can be shared amoung the tests.